20240927-multiple-object-deletion.md

```diff
@@ -0,0 +1,163 @@
+tags: [ "gateway", "satellite", "satellite/metainfo", "uplink" ]
+---
+
+# Multiple Object Deletion
+
+## Essentials
+
+### Header
+
+Date: 2024-09-27
+
+Owner: Jeremy Wharton
+
+Accountable:
+- Edge team
+
+Consulted:
+- Product team
+
+Informed:
+- QA team
+
+### Context
+
+The satellite metainfo RPC service provides an API through which objects stored on the Storj network may be interacted with. While this service exposes an endpoint through which individual objects may be deleted, it lacks an endpoint for deleting multiple objects in a single request. This limitation impacts services that depend on the metainfo RPC service, such as our S3-compatible gateway, by requiring separate requests for each object in a bulk deletion operation.
+
+This approach has several issues:
+
+- Network overhead: The amount of time spent transmitting and processing network requests increases with the number of objects to be deleted. This not only delays bulk operations but also consumes unnecessary bandwidth.
+
+- Rate limiting: Each deletion request counts against the deletion rate limit of the project associated with the request. This reduces the amount of deletion operations that can be performed within a given time frame. Additionally, this may cause bulk deletions to fail partway if the rate limit is reached mid-operation.
```

Comment:
I was little unclear on what we're intended to do with rate limit: how would this new endpoint handle rate limits? It might be worth describing it a bit more in a new section, perhaps?

Reply:
I added a sentence clarifying the new rate limiting behavior. Please let me know if it's sufficient or if you'd like more information to be added.

Reply:
That looks good. Thank you!